### PR TITLE
Backports v1.4 15-02-2019

### PR DESCRIPTION
**Pull request by eloycoto**
#6660
#6886
#7048
#7065
#7076
#7078
#7082
#6651
#7147

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7108)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-missed-k8s

**Comment by ianvernon** [comment]
test-docs-please

File: examples/kubernetes/1.10/cilium-containerd-ds.yaml [hunk 1]
```diff
@@ -175,7 +175,7 @@ spec:
               key: auto-direct-node-routes
               name: cilium-config
               optional: true
-        image: docker.io/cilium/cilium:v1.4.0
+        image: docker.io/cilium/cilium:1.4
```
Comment (ianvernon) [inline examples/kubernetes/1.10/cilium-containerd-ds.yaml]: @eloycoto this should not have been changed - it should still be `v1.4.0`.

**Review by ianvernon — CHANGES_REQUESTED** [review]

File: examples/kubernetes/1.10/cilium-containerd-ds.yaml (hunk 1 above)
Comment (eloycoto) [reply examples/kubernetes/1.10/cilium-containerd-ds.yaml]: is already fixed ;-)

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-missed-k8s

**Comment by eloycoto** [comment]
Found a panic in the log: 

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1683/

Panic: 
```
2019-02-17T08:54:14.157113276Z panic: runtime error: invalid memory address or nil pointer dereference
2019-02-17T08:54:14.15726075Z [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1a97437]
2019-02-17T08:54:14.157296724Z 
2019-02-17T08:54:14.157378582Z goroutine 914 [running]:
2019-02-17T08:54:14.157468566Z github.com/cilium/cilium/pkg/kvstore/allocator.(*Allocator).ForeachCache(0x0, 0xc000d0b3d0)
2019-02-17T08:54:14.157564894Z 	/go/src/github.com/cilium/cilium/pkg/kvstore/allocator/allocator.go:363 +0x37
2019-02-17T08:54:14.157622759Z github.com/cilium/cilium/pkg/identity/cache.GetIdentityCache(0x2a018c0)
2019-02-17T08:54:14.157716935Z 	/go/src/github.com/cilium/cilium/pkg/identity/cache/cache.go:51 +0x73
2019-02-17T08:54:14.157876331Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regeneratePolicy(0xc0003ee780, 0x30241a0, 0xc0004d46c0, 0x0, 0x0)
2019-02-17T08:54:14.157995122Z 	/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:123 +0xca
2019-02-17T08:54:14.15817589Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).runPreCompilationSteps(0xc0003ee780, 0x30241a0, 0xc0004d46c0, 0xc00010be00, 0x0, 0x0)
2019-02-17T08:54:14.15826263Z 	/go/src/github.com/cilium/cilium/pkg/endpoint/bpf.go:762 +0xbe5
2019-02-17T08:54:14.158524106Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regenerateBPF(0xc0003ee780, 0x30241a0, 0xc0004d46c0, 0xc00010be00, 0x0, 0x0, 0x0, 0x0)
2019-02-17T08:54:14.15863281Z 	/go/src/github.com/cilium/cilium/pkg/endpoint/bpf.go:537 +0x16f
2019-02-17T08:54:14.158844662Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regenerate(0xc0003ee780, 0x30241a0, 0xc0004d46c0, 0xc00010be00, 0x0, 0x0)
2019-02-17T08:54:14.158938316Z 	/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:307 +0x703
2019-02-17T08:54:14.15909576Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).Regenerate.func1(0xc00102a920, 0xc000147500, 0xc0003ee780, 0x30241a0, 0xc0004d46c0)
2019-02-17T08:54:14.159187905Z 	/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:432 +0x230
2019-02-17T08:54:14.159234396Z created by github.com/cilium/cilium/pkg/endpoint.(*Endpoint).Regenerate
2019-02-17T08:54:14.159329203Z 	/go/src/github.com/cilium/cilium/pkg/endpoint/policy.go:407 +0x89
2019-02-17T08:54:07.089271018Z level=debug msg="os.Hostname() returned" nodeName=k8s1 subsys=node
2019-02-17T08:54:07.155723249Z level=info msg="  --access-log=''" subsys=daemon
```

**Comment by eloycoto** [comment]
@cilium/janitors IMHO, the issue with the panic is the lack of this PR in 1.4: 

https://github.com/cilium/cilium/pull/6651/files

the current code miss the If, so I understand that the iussue is in there. 
https://github.com/cilium/cilium/blob/c71dafa2efb360ee5de35b153ded171757144231/pkg/identity/cache/cache.go#L48-L71

**Comment by eloycoto** [comment]
test-missed-k8s

**Comment by eloycoto** [comment]
Added PR: 

#7110
#7103

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-missed-k8s

File: examples/kubernetes/1.10/cilium-containerd-ds.yaml (hunk 1 above)
Comment (tgraf) [reply examples/kubernetes/1.10/cilium-containerd-ds.yaml]: How is it fixed?

Comment (eloycoto) [reply examples/kubernetes/1.10/cilium-containerd-ds.yaml]: Comment is outdated: 
https://github.com/cilium/cilium/pull/7108/files#diff-bb04b0a781aef2e330d7f84c05d4b78fR343

**Comment by eloycoto** [comment]
Added 

#6651
#7147

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-missed-k8s

**Review by ianvernon — APPROVED** [review]

**Pull request merged**